src/xrpld/app/tx/detail/Transactor.cpp

```diff
@@ -247,6 +300,21 @@ TER
 Transactor::payFee()
 {
     auto const feePaid = ctx_.tx[sfFee].xrp();
+    if (ctx_.tx.isDelegated())
```

Comment:
Why is this needed? We  always need the account sender to update the balance. I.e. it's always `sfAccount`. And if it's not delegated then `account_` is `sfAccount`.